modules/collect/src/main/java/com/opengamma/strata/collect/result/FailureItem.java

```diff
@@ -112,7 +119,25 @@ static FailureItem of(FailureReason reason, String message, int skipFrames) {
     ArgChecker.notNull(reason, "reason");
     ArgChecker.notEmpty(message, "message");
     String stackTrace = localGetStackTraceAsString(message, skipFrames);
-    return new FailureItem(reason, message, stackTrace, null);
+    return new FailureItem(reason, message, ImmutableMap.of(), stackTrace, null);
+  }
+
+  /**
+   * Obtains a failure from a reason and message.
+   * <p>
+   * The failure will still have a stack trace, but the cause type will not be present.
+   * 
+   * @param reason  the reason
+   * @param message  the failure message, not empty
+   * @param attributes the attributes associated to this failure
+   * @param skipFrames  the number of caller frames to skip, not including this one
+   * @return the failure
+   */
+  static FailureItem of(FailureReason reason, String message, Map<String, String> attributes, int skipFrames) {
```

Comment:
This is package-private. I'm assuming you want it to be public. If so you shouldn't have the `skipFrames` parameter.